### DISCUSSION TITLE
How to stream video in range ?

### DISCUSSION THREAD

**Discussion by anboo44** [activej/activej#301] [Q&A] [answered]
It is ok to play. However, I cannot scroll video on timeline. My client uses html with video tag
Please support range request when returning file

![287600876-654d7a4f-5913-4b9a-b92d-83b17ecd8ba9](https://github.com/activej/activej/assets/19175021/1f407013-188c-4f5c-8a33-867aa74e3657)

**Comment by anboo44** [comment]
@eduard-vasinskyi here

**Comment by eduard-vasinskyi** [comment, marked as answer]
Hi, @anboo44

ActiveJ v5.x `HttpResponse` had built-in support for [downloading files using range requests.](https://github.com/activej/activej/blob/aa68c6bb9cc179c3441fea928dea90b9098c0df0/core-http/src/main/java/io/activej/http/HttpResponse.java#L190-L249). In v6.0 we cleaned up API in ActiveJ HTTP. As a result, we moved the method directly to [FileSystemServlet](https://github.com/activej/activej/blob/v6.0-beta2/cloud-fs/src/main/java/io/activej/fs/http/FileSystemServlet.java) as it was too specific for ActiveJ HTTP. 

You can copy the code or parts of it that meet your requirements and use that to create your own method that serves files by range.
 
We have actually used ActiveJ HTTP in the past to develop a web video player (with range requests support).

**Reply by anboo44** [reply]
@eduard-vasinskyi Thank you so much
